GitOut/Features/Storage/FileStorage.cs

```diff
@@ -10,13 +10,16 @@ public class FileStorage : IWritableStorage
         public void Write(string key, object value)
         {
             string configFile = SettingsOptions.GetSettingsPath();
-            Directory.CreateDirectory(Directory.GetParent(configFile).FullName);
+            if (Directory.GetParent(configFile) is DirectoryInfo existing)
+            {
+                Directory.CreateDirectory(existing.FullName);
+            }
 
             IDictionary<string, object> sections;
             try
             {
                 string text = File.ReadAllText(configFile);
-                sections = JsonSerializer.Deserialize<IDictionary<string, object>>(text);
+                sections = JsonSerializer.Deserialize<IDictionary<string, object>>(text) ?? new Dictionary<string, object>();
```

Comment:
Not a 100% on this. In what circumstance would Deserialize return null, and do we want to clear all previous settings in that case?
Looking at IOException below may be to wide as well, we probably only want to initialize a new list if we get a FileNotFoundException? 🤔